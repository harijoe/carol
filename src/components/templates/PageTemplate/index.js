import React, { PropTypes } from 'react'
import styled, { ThemeProvider } from 'styled-components'
import NotificationsSystem from 'reapop'
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider'
import theme from 'reapop-theme-wybo'
import { addLocaleData } from 'react-intl'
import fr from 'react-intl/locale-data/fr'
import en from 'react-intl/locale-data/en'
import es from 'react-intl/locale-data/es'

import { injectGlobals } from 'utils/style'
import defaultTheme, { resets, scaffolding } from '../../themes/default'

injectGlobals([resets, scaffolding])

addLocaleData([...es, ...en, ...fr])

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  padding-top: 3.75rem;
  min-height: 100vh;
  box-sizing: border-box;
`

const Header = styled.header`
  position: fixed;
  top: 0;
  width: 100%;
  z-index: 999;
`

const Content = styled.section`
  width: 100%;
  box-sizing: border-box;
  margin: 2rem auto;
  max-width: 920px;
`

const Footer = styled.footer`
  margin-top: auto;
`

const notificationsDefaultValues = {
  status: 'info',
  position: 'tr',
  dismissible: true,
  dismissAfter: 5000,
  allowHTML: true,
  closeButton: false,
}

const PageTemplate = ({ header, children, footer, ...props }) => (
  <MuiThemeProvider>
    <ThemeProvider theme={defaultTheme}>
      <Wrapper {...props}>
        <NotificationsSystem theme={theme} defaultValues={notificationsDefaultValues} />
        <Header {...props}>{header}</Header>
        <Content>{children}</Content>
        <Footer>{footer}</Footer>
      </Wrapper>
    </ThemeProvider>
  </MuiThemeProvider>
)

PageTemplate.propTypes = {
  header: PropTypes.any.isRequired,
  children: PropTypes.any.isRequired,
  footer: PropTypes.any.isRequired,
}

export default PageTemplate
