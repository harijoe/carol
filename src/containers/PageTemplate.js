import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { fromContext } from 'store/selectors'

import { PageTemplate } from 'components'

const PageTemplateContainer = props =>
  <PageTemplate {...props}>
    {props.children}
  </PageTemplate>

PageTemplateContainer.propTypes = {
  children: PropTypes.any,
}

const mapStateToProps = state => ({
  ssr: fromContext.isSSR(state),
})

export default connect(mapStateToProps)(PageTemplateContainer)
