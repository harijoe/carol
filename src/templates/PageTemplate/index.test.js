import React from 'react'
import { Provider } from 'react-redux'
import { shallow } from 'enzyme'
import mockStore from 'mocks/storeMock'
import PageTemplate from '.'

const store = mockStore()

const wrapper = shallow(
  <Provider store={store}>
    <PageTemplate header="header" footer="footer">
      test
    </PageTemplate>
  </Provider>,
)

window.location.hostname = 'carol-co-uk.dev.quotatis.net'

it('renders PageTemplate', () => {
  expect(wrapper).toMatchSnapshot()
  expect(wrapper.find(PageTemplate)).toHaveLength(1)
})
