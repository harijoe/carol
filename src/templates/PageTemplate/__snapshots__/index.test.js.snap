// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PageTemplate 1`] = `
<PageTemplate
  footer="footer"
  header="header"
>
  test
</PageTemplate>
`;
